File: baseLibrary/src/main/java/com/base/baselibrary/base/ImmersionBarUtil.java
package com.base.baselibrary.base;

import android.app.Activity;

import androidx.annotation.ColorRes;
import androidx.fragment.app.Fragment;

import com.gyf.immersionbar.ImmersionBar;

/**
 * Create by Darren
 * On 2020/10/9 12:18
 * 沉浸式工具类
 **/
public class ImmersionBarUtil {
    public static void initImmersionBar(Activity activity, StatusBarMode statusBarMode, @ColorRes int statusBarColor, boolean statusBarDarkFont, boolean fixKeyboard) {
        ImmersionBar immersionBar = ImmersionBar.with(activity);
        switch (statusBarMode) {
            case DEFAULT:
                immersionBar.fitsSystemWindows(true);
                immersionBar.statusBarColor(statusBarColor);
                break;
            case IMMEPIC:
                immersionBar.transparentStatusBar();
                break;
        }
        immersionBar.statusBarDarkFont(statusBarDarkFont);
//        immersionBar.keyboardEnable(fixKeyboard);
        immersionBar.init();
    }

    public static void initImmersionBar(Fragment fragment, StatusBarMode statusBarMode, @ColorRes int statusBarColor, boolean statusBarDarkFont, boolean fixKeyboard) {
        ImmersionBar immersionBar = ImmersionBar.with(fragment);
        switch (statusBarMode) {
            case DEFAULT:
                immersionBar.fitsSystemWindows(true);
                immersionBar.statusBarColor(statusBarColor);
                break;
            case IMMEPIC:
                immersionBar.transparentStatusBar();
                break;
        }
        immersionBar.statusBarDarkFont(statusBarDarkFont);
//        immersionBar.keyboardEnable(fixKeyboard);
        immersionBar.init();
    }
}

File: baseLibrary/src/main/java/com/base/baselibrary/basekt/ImmersionBarUtil.kt
package com.base.baselibrary.basekt

import android.app.Activity
import androidx.annotation.ColorRes
import androidx.fragment.app.Fragment
import com.base.baselibrary.base.StatusBarMode
import com.gyf.immersionbar.ImmersionBar
import com.gyf.immersionbar.ktx.immersionBar

/**
 * Create by Darren
 * On 2020/10/9 12:18
 * 沉浸式工具类
 */
fun Activity.initImmersionBar(
    statusBarMode: StatusBarMode?,
    @ColorRes statusBarColor: Int,
    statusBarDarkFont: Boolean
) {
    immersionBar {
        when (statusBarMode) {
            StatusBarMode.DEFAULT -> {
                fitsSystemWindows(true)
                statusBarColor(statusBarColor)
            }
            StatusBarMode.IMMEPIC -> {
                transparentStatusBar()
            }
        }
        statusBarDarkFont(statusBarDarkFont)
    }
}

fun Fragment.initImmersionBar(
    statusBarMode: StatusBarMode?,
    @ColorRes statusBarColor: Int,
    statusBarDarkFont: Boolean
) {
    immersionBar {
        when (statusBarMode) {
            StatusBarMode.DEFAULT -> {
                fitsSystemWindows(true)
                statusBarColor(statusBarColor)
            }
            StatusBarMode.IMMEPIC -> {
                transparentStatusBar()
            }
        }
        statusBarDarkFont(statusBarDarkFont)
    }
}
